pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/ReportManager.kt

```diff
@@ -251,53 +246,52 @@ class ReportManager: KoinComponent {
             this.dataStreamId = dataStreamId
             this.dataStreamRoute = dataStreamRoute
             this.dexIngestDateTime = dexIngestDateTime
-            this.messageMetadata = messageMetadata
-            this.stageInfo= stageInfo
+            this.jurisdiction = jurisdiction
+            this.senderId = senderId
+            this.messageMetadata = messageMetadata?.toMessageMetadata()
+            this.stageInfo = stageInfo?.toStageInfo()
             this.tags = tags
             this.data = data
             this.jurisdiction = jurisdiction
             this.senderId = senderId
             this.dataProducerId= dataProducerId
-            this.source = source
+            this.source = source.toString()
             this.contentType = contentType
-
-            if (contentType.lowercase() == "json") {
-                val typeObject = object : TypeToken<HashMap<*, *>?>() {}.type
-                val jsonMap: Map<String, Any> = gson.fromJson(Gson().toJson(content, MutableMap::class.java).toString(), typeObject)
-                this.content = jsonMap
-            } else
-                this.content = content
+            this.content = getContent(contentType, content)
         }
-       return createReportItem(uploadId,stageReportId,stageReport)
+       return createReportItem(uploadId, stageReportId, stageReport)
     }
 
     /**
      * Creates a dead-letter report if there is a malformed data or missing required fields
      *
-     * @param uploadId String
-     * @param dataStreamId String
-     * @param dataStreamRoute String
-     * @param messageMetadata MessageMetadata?
-     * @param stageInfo StageInfo?
-     * @param tags Map<String,String>??
-     * @param data Map<String,String>?
-     * @param contentType String
-     * @param content String
+     * @param uploadId String?
+     * @param dataStreamId String?
+     * @param dataStreamRoute String?
+     * @param dexIngestDateTime Instant?
+     * @param messageMetadata MessageMetadataSB?
+     * @param stageInfo StageInfoSB?
+     * @param tags Map<String, String>?
+     * @param data Map<String, String>?
+     * @param dispositionType DispositionType
+     * @param contentType String?
+     * @param content Any?
      * @param jurisdiction String?
      * @param senderId String?
+     * @param deadLetterReasons List<String>
+     * @param validationSchemaFileNames List<String>
      * @param dataProducerId String?
      * @return String
      * @throws BadStateException
      */
-
     @Throws(BadStateException::class)
     fun createDeadLetterReport(
         uploadId: String?,
         dataStreamId: String?,
         dataStreamRoute: String?,
-        dexIngestDateTime: Date?,
-        messageMetadata: MessageMetadata?,
-        stageInfo: StageInfo?,
+        dexIngestDateTime: Instant?,
+        messageMetadata: MessageMetadataSB?,
+        stageInfo: StageInfoSB?,
```

Comment:
should we remove SB (Service Bus reference) from the following? `MessageMetadataSB` and `StageInfoSB`?